federation/pkg/federation-controller/secret/secret_controller.go

```diff
@@ -55,38 +55,40 @@ var (
 	RequiredResources = []schema.GroupVersionResource{apiv1.SchemeGroupVersion.WithResource("secrets")}
 )
 
-type SecretController struct {
-	// For triggering single secret reconciliation. This is used when there is an
-	// add/update/delete operation on a secret in either federated API server or
-	// in some member of the federation.
-	secretDeliverer *util.DelayingDeliverer
-
-	// For triggering all secrets reconciliation. This is used when
-	// a new cluster becomes available.
+type Synchronizer struct {
+	// For triggering reconciliation of a single resource. This is
+	// used when there is an add/update/delete operation on a resource
+	// in either federated API server or in some member of the
+	// federation.
+	deliverer *util.DelayingDeliverer
```

Comment:
Not necessary in this PR, but I'd like to see very descriptive comments on all of these objects. It was a little hard for me to understand what was going on here when I joined the team, and having descriptive comments on these would have been really helpful.